core/src/test/java/org/apache/accumulo/core/clientImpl/ScannerBaseTest.java

```diff
@@ -20,71 +20,121 @@
 
 import static org.easymock.EasyMock.createMock;
 import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.expectLastCall;
+import static org.easymock.EasyMock.partialMockBuilder;
 import static org.easymock.EasyMock.replay;
 import static org.easymock.EasyMock.verify;
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
 
+import java.awt.*;
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.Map;
+import java.util.Scanner;
+import java.util.concurrent.TimeUnit;
 import java.util.function.BiConsumer;
 
+import org.apache.accumulo.core.client.AccumuloClient;
+import org.apache.accumulo.core.client.IteratorSetting;
 import org.apache.accumulo.core.client.ScannerBase;
+import org.apache.accumulo.core.client.sample.SamplerConfiguration;
 import org.apache.accumulo.core.data.Key;
 import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.security.Authorizations;
 import org.apache.hadoop.io.Text;
+import org.apache.hadoop.shaded.org.mockito.Mockito;
+import org.easymock.EasyMock;
 import org.junit.Test;
 
 public class ScannerBaseTest {
 
-  private static class forEachTester {
-
-    private Map<Key,Value> map;
-
-    forEachTester(Map<Key,Value> map) {
-      this.map = map;
-    }
-
-    public void forEach(BiConsumer<? super Key,? super Value> keyValueConsumer) {
-      for (Map.Entry<Key,Value> entry : this.map.entrySet()) {
-        keyValueConsumer.accept(entry.getKey(), entry.getValue());
-      }
-    }
-  }
-
   @Test
   public void testScannerBaseForEach() throws Exception {
 
-    ScannerBase s;
-    Map<Key,Value> scannerMap;
-    Iterator<Map.Entry<Key,Value>> it;
-    Key key;
-    Value val;
-    BiConsumer<Key,Value> keyValueConsumer;
-    forEachTester fet;
+    //This subclass of ScannerBase contains a List that ScannerBase.forEach() can
+    //iterate over for testing purposes.
+    class MockScanner extends List implements ScannerBase {
```

Comment:
Extending List forces you to add implementations for a bunch of methods. You can just implement ScannerBase, and have a constructor that takes a `map`, which itself is easily constructed, using `Map.of(...)`. MockScanner's `iterator()` method can just return `map.entrySet().iterator()`.